IdentityCore/src/requests/MSIDSilentTokenRequest.m

```diff
@@ -235,129 +216,148 @@ - (void)executeRequestImpl:(MSIDRequestCompletionBlock)completionBlock
+                NSError *interactionError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInteractionRequired, @"User interaction is required", error.msidOauthError, error.msidSubError, error, self.requestParameters.correlationId, nil, YES);
+                completionBlock(nil, interactionError);
+            }
+            else
+            {
+                completionBlock(nil, error);
+            }
+        }
+        else
+        {
+            completionBlock(result, nil);
+        }
+    }];
 }
 
-- (void)tryAppRefreshToken:(MSIDBaseToken<MSIDRefreshableToken> *)multiResourceRefreshToken
-           completionBlock:(nonnull MSIDRequestCompletionBlock)completionBlock
+#pragma mark - Helpers
+
+- (MSIDBaseToken<MSIDRefreshableToken> *)handleErrorResponseForAppRefreshToken:(MSIDBaseToken<MSIDRefreshableToken> *)refreshToken
 {
-    if (!multiResourceRefreshToken)
+    NSError *error = nil;
+    MSIDRefreshToken *familyRefreshToken = [self familyRefreshTokenWithError:&error];
+    if (error)
     {
-        MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Refresh token wasn't found, user interaction is required.");
-        
-        NSError *interactionError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInteractionRequired, @"User interaction is required", nil, nil, nil, self.requestParameters.correlationId, nil, YES);
-        completionBlock(nil, interactionError);
-        return;
+        MSID_LOG_WITH_CTX_PII(MSIDLogLevelError, self.requestParameters, @"Failed to retrieve Family Refresh token with error %@, and user interaction is required", MSID_PII_LOG_MASKABLE(error));
+        return nil;
+    }
+
+    if (familyRefreshToken && ![[familyRefreshToken refreshToken] isEqualToString:[refreshToken refreshToken]])
+    {
+        return familyRefreshToken;
     }
     
-    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Found Refresh token, using it...");
-    [self refreshAccessToken:multiResourceRefreshToken
-             completionBlock:^(MSIDTokenResult * _Nullable result, NSError * _Nullable error) {
-                 if (error)
-                 {
-                     //Check if server returns invalid_grant or invalid_request
-                     if ([self isErrorRecoverableByUserInteraction:error])
-                     {
-                         NSError *interactionError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInteractionRequired, @"User interaction is required", error.msidOauthError, error.msidSubError, error, self.requestParameters.correlationId, nil, YES);
-
-                         completionBlock(nil, interactionError);
-                         return;
-                     }
-                     else
-                     {
-                         completionBlock(nil, error);
-                     }
-                 }
-                 else
-                 {
-                     completionBlock(result, nil);
-                 }
-             }];
+    return nil;
```

Comment:
Seems nobody invokes completion block in this case?

Reply:
Could you specify which completion block?